### PR TITLE
specs for choose programme advisory page

### PR DESCRIPTION
**Pull request by goodviber**
### Context
Trello #486
We need to have a speed bump advisory page prior to the /schools/choose-programme page 
### Changes proposed in this pull request
controller method, route and view for /schools/choose-programme/advisory
### Guidance to review
Login as induction coordinator without having chosen a programme....navigate to /schools/choose-programme/advisory
### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

**Comment by goodviber** [comment]
[Dashboard: school not in partnership or not chosen programme yet](https://trello.com/c/2LvmwlB6/486-dashboard-school-not-in-partnership-or-not-chosen-programme-yet)

File: app/views/schools/choose_programme/advisory.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<% content_for :before_content, govuk_back_link(
+  text: "Back",
+  href: root_url) # what is the back path here, could we use an internal referer helper?
+%>
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
```
Comment (kicferk1) [inline app/views/schools/choose_programme/advisory.html.erb:6]: Let's add closing tags for these?

File: app/views/schools/choose_programme/advisory.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      <p class="govuk-body">You will be choosing a programme for any new early career teachers. They will complete this programme over 2 years.</p>
+      <p class="govuk-body">Only use this service if your school wants to:</p>
+      <ul class="govuk-list govuk-list--bullet">
+        <li>use an approved training provider</li>
+      </ul>
+      <p class="govuk-body">Or</p>
+      <ul class="govuk-list govuk-list--bullet">
+        <li>use the DfE accredited training materials</li>
+      </ul>
+      <p class="govuk-body">Not sure> Check our guidance on <%= govuk_link_to "induction programmes for early career teacher.", "https://www.gov.uk/government/publications/early-career-framework-reforms-overview/early-career-framework-reforms-overview" %></p>
```
Comment (ethanmills) [inline app/views/schools/choose_programme/advisory.html.erb:17]: Typo

File: app/views/schools/choose_programme/advisory.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<% content_for :before_content, govuk_back_link(
+  text: "Back",
+  href: root_url) # what is the back path here, could we use an internal referer helper?
+%>
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <h1 class="govuk-heading-xl">Do you know how you want to run your statutory induction programme?</h1>
+      <p class="govuk-body">You will be choosing a programme for any new early career teachers. They will complete this programme over 2 years.</p>
+      <p class="govuk-body">Only use this service if your school wants to:</p>
+      <ul class="govuk-list govuk-list--bullet">
```
Comment (kicferk1) [inline app/views/schools/choose_programme/advisory.html.erb:10]: `p`s and `ul` are not children of `h1`. Can we bring them 1 level of indentation to the left?

File: app/views/schools/choose_programme/advisory.html.erb (hunk 2 above)
Comment (goodviber) [reply app/views/schools/choose_programme/advisory.html.erb:17]: ooooops

File: app/views/schools/choose_programme/advisory.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<% content_for :before_content, govuk_back_link(
+  text: "Back",
+  href: root_url) # what is the back path here, could we use an internal referer helper?
+%>
```
Comment (ethanmills) [inline app/views/schools/choose_programme/advisory.html.erb:4]: No back link on designs, confirmed with Al, should not be here

File: app/views/schools/choose_programme/advisory.html.erb (hunk 1 above)
Comment (goodviber) [reply app/views/schools/choose_programme/advisory.html.erb:6]: It's been a long day

File: app/views/schools/choose_programme/advisory.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    <h1 class="govuk-heading-xl">Do you know how you want to run your statutory induction programme?</h1>
+      <p class="govuk-body">You will be choosing a programme for any new early career teachers. They will complete this programme over 2 years.</p>
+      <p class="govuk-body">Only use this service if your school wants to:</p>
+      <ul class="govuk-list govuk-list--bullet">
+        <li>use an approved training provider</li>
+      </ul>
+      <p class="govuk-body">Or</p>
+      <ul class="govuk-list govuk-list--bullet">
+        <li>use the DfE accredited training materials</li>
+      </ul>
```
Comment (ethanmills) [inline app/views/schools/choose_programme/advisory.html.erb]: Is the markup here definitely right? This doesn't feel like two lists semantically, and I wonder how a screenreader would deal with this

File: app/controllers/schools/choose_programme_controller.rb [hunk 6]
```diff
@@ -3,15 +3,12 @@
...
   skip_after_action :verify_policy_scoped
+  before_action :verify_programme_chosen, only: %i[advisory show]
 
-  def show
-    school = current_user.induction_coordinator_profile.schools.first
+  def advisory; end
 
-    if school.chosen_programme?(Cohort.current)
-      redirect_to helpers.profile_dashboard_url(current_user)
-    else
```
Comment (ethanmills) [inline app/controllers/schools/choose_programme_controller.rb:12]: Why has this been removed?

File: app/views/schools/choose_programme/advisory.html.erb (hunk 5 above)
Comment (goodviber) [reply app/views/schools/choose_programme/advisory.html.erb]: LGTM, open to suggestions ?

File: app/views/schools/choose_programme/advisory.html.erb [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<div class="govuk-grid-row">
```
Comment (ethanmills) [inline app/views/schools/choose_programme/advisory.html.erb:1]: Page title please, "Check guidance" per https://docs.google.com/document/d/11ubyUyMMz7OSF1Tu1JT6Ehg38l31vOfT-hZUcUTOFOM/edit#heading=h.f6u8iwmrtpft

File: app/views/schools/choose_programme/show.html.erb [hunk 8]
```diff
@@ -1,3 +1,7 @@
+<% content_for :before_content, govuk_back_link(
+  text: "Back",
+  href: advisory_schools_choose_programme_path)
+%>
 <div class="govuk-grid-row">
```
Comment (ethanmills) [inline app/views/schools/choose_programme/show.html.erb:5]: Could you add the title for this one too please? "Choose programme"

File: app/views/schools/choose_programme/advisory.html.erb (hunk 5 above)
Comment (callumacrae) [reply app/views/schools/choose_programme/advisory.html.erb]: I would go with:

```
      <ul class="govuk-list govuk-list--bullet">
        <li>use an approved training provider; or</li>
        <li>use the DfE accredited training materials</li>
      </ul>
```

However, that's a copy change which would probably require approval from whoever wrote the copy in the first place

Comment (goodviber) [reply app/views/schools/choose_programme/advisory.html.erb]: open to suggestions? LGTM

**Review by kicferk1 — APPROVED** [review]

**Review by amazimbe — APPROVED** [review]
LGTM!

**Comment by ethanmills** [comment]
[Dashboard: school not in partnership or not chosen programme yet](https://trello.com/c/2LvmwlB6/486-dashboard-school-not-in-partnership-or-not-chosen-programme-yet)

**Pull request merged**